### PR TITLE
Update `ConciseView` to remove unnecessary text and not color entire line in red

### LINKED ISSUE #10716: ConciseView for $ErrorView does not work in StrictMode
# Steps to reproduce

```powershell
> Set-StrictMode -Version Latest
> 1/0
>
> $error[0]|fl *

InvalidOperation: The variable '$_CategoryInfo' cannot be retrieved because it has not been set.
>
> $error[0]|fl * -f

PSMessageDetails      :
Exception             : System.Management.Automation.RuntimeException: The variable '$_CategoryInfo' cannot be retrieved because it has
                        not been set.
                           at System.Management.Automation.ExceptionHandlingOps.CheckActionPreference(FunctionContext funcContext,
                        Exception exception) in D:\Users\Bukem\Documents\_DEVELOPMENT\powershell\src\System.Management.Automation\engine\
                        runtime\Operations\MiscOps.cs:line 1635
                           at System.Management.Automation.Interpreter.ActionCallInstruction`2.Run(InterpretedFrame frame) in D:\Users\Bu
                        kem\Documents\_DEVELOPMENT\powershell\src\System.Management.Automation\engine\interpreter\CallInstruction.Generat
                        ed.cs:line 504
                           at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame) in D:\
                        Users\Bukem\Documents\_DEVELOPMENT\powershell\src\System.Management.Automation\engine\interpreter\ControlFlowInst
                        ructions.cs:line 391
                           at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame) in D:\
                        Users\Bukem\Documents\_DEVELOPMENT\powershell\src\System.Management.Automation\engine\interpreter\ControlFlowInst
                        ructions.cs:line 357
                           at System.Management.Automation.Interpreter.Interpreter.Run(InterpretedFrame frame) in D:\Users\Bukem\Document
                        s\_DEVELOPMENT\powershell\src\System.Management.Automation\engine\interpreter\Interpreter.cs:line 104
                           at System.Management.Automation.Interpreter.LightLambda.RunVoid1[T0](T0 arg0) in D:\Users\Bukem\Documents\_DEV
                        ELOPMENT\powershell\src\System.Management.Automation\engine\interpreter\LightLambda.Generated.cs:line 82
                           at System.Management.Automation.ScriptBlock.InvokeWithPipeImpl(ScriptBlockClauseToInvoke clauseToInvoke,
                        Boolean createLocalScope, Dictionary`2 functionsToDefine, List`1 variablesToDefine, ErrorHandlingBehavior
                        errorHandlingBehavior, Object dollarUnder, Object input, Object scriptThis, Pipe outputPipe, InvocationInfo
                        invocationInfo, Object[] args) in D:\Users\Bukem\Documents\_DEVELOPMENT\powershell\src\System.Management.Automati
                        on\engine\runtime\CompiledScriptBlock.cs:line 1188
                           at System.Management.Automation.ScriptBlock.InvokeWithPipe(Boolean useLocalScope, ErrorHandlingBehavior
                        errorHandlingBehavior, Object dollarUnder, Object input, Object scriptThis, Pipe outputPipe, InvocationInfo
                        invocationInfo, Boolean propagateAllExceptionsToTop, List`1 variablesToDefine, Dictionary`2 functionsToDefine,
                        Object[] args) in D:\Users\Bukem\Documents\_DEVELOPMENT\powershell\src\System.Management.Automation\engine\lang\s
                        criptblock.cs:line 1043
                           at System.Management.Automation.ScriptBlock.DoInvokeReturnAsIs(Boolean useLocalScope, ErrorHandlingBehavior
                        errorHandlingBehavior, Object dollarUnder, Object input, Object scriptThis, Object[] args) in D:\Users\Bukem\Docu
                        ments\_DEVELOPMENT\powershell\src\System.Management.Automation\engine\lang\scriptblock.cs:line 927
                           at Microsoft.PowerShell.Commands.PSPropertyExpression.GetValue(PSObject target, Boolean eatExceptions) in D:\U
                        sers\Bukem\Documents\_DEVELOPMENT\powershell\src\System.Management.Automation\FormatAndOutput\common\Utilities\Ms
                        hexpression.cs:line 309
TargetObject          : _CategoryInfo
CategoryInfo          : InvalidOperation: (_CategoryInfo:String) [], RuntimeException
FullyQualifiedErrorId : VariableIsUndefined
ErrorDetails          :
InvocationInfo        : System.Management.Automation.InvocationInfo
ScriptStackTrace      : at Get-ConciseViewPositionMessage, <No file>: line 174
                        at <ScriptBlock>, <No file>: line 190
                        at <ScriptBlock>, <No file>: line 1
PipelineIterationInfo : {}
```

# Expected behavior

```powershell
> Set-StrictMode -Version Latest
> 1/0
Error: Attempted to divide by zero.
```

# Actual behavior

```powershell
> Set-StrictMode -Version Latest
> 1/0
>
```

# Environment data

```powershell
Name                           Value
----                           -----
PSVersion                      7.0.0-preview.4
PSEdition                      Core
GitCommitId                    7.0.0-preview.4-37-g8cf9c018000d9643328edf3d66ddfc52fe913950
OS                             Microsoft Windows 10.0.18362
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

```

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
# PR Summary

Remove unnecessary text before script path, detect if whole line is emphasized, then don't do any emphasis.  Change emphasis color to accent color instead of error color to make it easier to read.  Also fixed some typos and disabled strict mode in helper that works with different dynamic objects that puts errors in $error if strict mode is enabled.

Before:

img

After:

img

## PR Context

Fix https://github.com/PowerShell/PowerShell/issues/10716

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

**Comment by iSazonov** [comment]
Is it common practice to paint in red only an error token and keep comment in white-on-black?

**Comment by SteveL-MSFT** [comment]
@iSazonov the problem with that proposal is that some errors don't have a visible token, like `foreach ($a 123)` is missing `in` so nothing would be red.

**Comment by iSazonov** [comment]
'^' could be red in the case.

**Review by JamesWTruher — COMMENTED** [review]
nothing really blocking
I should have caught those other errors in the previous, sorry

File: src/System.Management.Automation/FormatAndOutput/DefaultFormatters/PowerShellCore_format_ps1xml.cs [hunk 1]
```diff
@@ -901,8 +902,13 @@ private static IEnumerable<FormatViewDefinition> ViewsOf_System_Management_Autom
                                             $line = $myinv.Line
                                             $highlightLine = $myinv.PositionMessage.Split('+').Count - 1
                                             $offsetLength = $myinv.PositionMessage.split('+')[$highlightLine].Trim().Length
-                                            $line = $line.Insert($myinv.OffsetInLine - 1 + $offsetLength, $resetColor)
-                                            $line = $line.Insert($myinv.OffsetInLine - 1, $errorColor)
+
+                                            # don't color the whole line red
+                                            if ($offsetLength -lt $line.Length - 1) {
+                                                $line = $line.Insert($myinv.OffsetInLine - 1 + $offsetLength, $resetColor)
```
[the remark this answers is not in the document]
Comment (SteveL-MSFT) [reply src/System.Management.Automation/FormatAndOutput/DefaultFormatters/PowerShellCore_format_ps1xml.cs]: True, will update

**Review by JamesWTruher — APPROVED** [review]

**Review by anmenaga — APPROVED** [review]

**Pull request merged**